test/integration/basic/basic_test.go

```diff
@@ -22,7 +21,8 @@ func TestBasic(t *testing.T) {
 		PublicClusters:  1,
 		PrivateClusters: 1,
 	}
-	testRunner.Build(t, needs, nil)
+	testRunner := &BasicTestRunner{}
```

Comment:
Maybe we don't even need a `BasicTestRunner` type, and instead
we could simply use the `ClusterTestRunnerBase`.

Same applies to the http and tcp_echo.

Reply:
I totally agree having a base class and "extending" was not a good idea. Propose to treat that in the next pr

Reply:
Ok.